Source/WebKit/Scripts/generate-umbrella-header.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+
+import filecmp
+import os
+import re
+
+
+def generate_umbrella_header(input_filename, output_filename, tempoutput_filename, header_dirname, framework_name):
+    include_pattern = re.compile(r"\#(include|import)\s+\<" + framework_name + r"\/(.*)\>")
+
+    existing_framework_headers = set()
+    output_headers = []
+    blocklist = {
+        "%sPrivate.h" % framework_name,
+        "%s.h" % framework_name,
+    }
+
+    with open(tempoutput_filename or output_filename, "w+") as output_file:
+        with open(input_filename, "r") as input_file:
+            for line in input_file:
+                match = include_pattern.search(line)
+                if match:
+                    existing_framework_headers.add(match.group(2))
+
+                output_file.write(line)
+
+        for (dirpath, dirnames, filenames) in os.walk(header_dirname):
```

Comment:
Agree that scanning the private headers is an anti-pattern, though the build system doesn't really give you a better option. The main issue here is that stale headers might be lurking in the build directory.

Reply:
This script comes from internal, so anything we do for WebKit should be done internally too.